packages/server/browser-automation-api/src/infrastructure/scraper/services/linkedin-automation-service.ts

```diff
@@ -37,12 +37,20 @@ export type Cookies = ReadonlyArray<{
   sameSite?: "Strict" | "Lax" | "None";
 }>;
 
+interface MessageDate {
+  year: number;
+  month?: number;
+  day?: number;
+  time: string;
+}
+
 export class LinkedinAutomationService {
   constructor(
     private cookies: Cookies,
     private userAgent: string,
     private proxyConfig: string,
   ) {}
+  private lastKnownTime: string | null = null;  // Add this as class property
```

Comment:
This service class should not have method related logic bound to it. This property should just be a variable declared inside the method it's used.